ledger/simulation/resources.go

```diff
@@ -0,0 +1,379 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package simulation
+
+import (
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// BoxRef is a reference to an application's box.
+type BoxRef struct {
+	// The actual ID of the app that owns the box.
+	App basics.AppIndex
+	// The name of the box.
+	Name string
+}
```

Comment:
This is the same thing as `logic.boxRef`, but that's not exported.

Reply:
I'd vote for exporting it.

Reply:
Or renaming it `BoxDescriptor`

Reply:
exported in 0861eec58062baa2b7da97deb78c16bd59d522ce